mlflow/server/js/src/experiment-tracking/components/artifact-view-components/ShowArtifactTextView.js

```diff
@@ -51,18 +51,31 @@ class ShowArtifactTextView extends Component {
       );
     } else {
       const language = getLanguage(this.props.path);
+      const renderedContent = ShowArtifactTextView.prettifyText(language, this.state.text);
       return (
         <div className='ShowArtifactPage'>
           <div className='text-area-border-box'>
             <SyntaxHighlighter language={language} style={style}>
-              {this.state.text}
+              {renderedContent}
             </SyntaxHighlighter>
           </div>
         </div>
       );
     }
   }
 
+  static prettifyText(language, rawText) {
+    if (language === 'json') {
```

Comment:
Since this function only changes the text when `language === 'json'`, would it make sense to move the JSON check to the function call? ie. `renderedContent = language === json ? ...prettifyJson(this.state.text) : this.state.text`

Reply:
Drive-by comment: I think the current structure is nice for future extensibility. If we add support for prettifying other text formats, users obtain that benefit automatically by continuing to use the same API (no need to add extra logic for `if language == yaml`, etc)

Reply:
sgtm :)